packages/contracts-bedrock/src/L1/OPStackManager.sol

```diff
@@ -155,34 +158,53 @@ contract OPStackManager is ISemver {
     /// focused on an OPSM version that unblocks interop, we are not proxying OPSM for simplicity.
     /// Later, we will `_disableInitializers` in the constructor and replace any constructor logic
     /// with an `initialize` function, which will be a breaking change to the OPSM interface.
-    constructor(SuperchainConfig _superchainConfig, ProtocolVersions _protocolVersions, Blueprints memory _blueprints) {
-        // TODO uncomment these as we add more validations to this contract, currently this will break a few tests.
-        // assertValidContractAddress(address(_superchainConfig));
-        // assertValidContractAddress(address(_protocolVersions));
-        // assertValidContractAddress(_blueprints.addressManager);
-        // assertValidContractAddress(_blueprints.proxy);
-        // assertValidContractAddress(_blueprints.proxyAdmin);
-        // assertValidContractAddress(_blueprints.l1ChugSplashProxy);
-        // assertValidContractAddress(_blueprints.resolvedDelegateProxy);
+    // constructor(SuperchainConfig _superchainConfig, ProtocolVersions _protocolVersions) {
+    //     // TODO uncomment these as we add more validations to this contract, currently this will break a few tests.
+    //     // assertValidContractAddress(address(_superchainConfig));
+    //     // assertValidContractAddress(address(_protocolVersions));
+    //     // assertValidContractAddress(_blueprints.addressManager);
+    //     // assertValidContractAddress(_blueprints.proxy);
+    //     // assertValidContractAddress(_blueprints.proxyAdmin);
+    //     // assertValidContractAddress(_blueprints.l1ChugSplashProxy);
+    //     // assertValidContractAddress(_blueprints.resolvedDelegateProxy);
+
+    //     superchainConfig = _superchainConfig;
+    //     protocolVersions = _protocolVersions;
+    // }
+    constructor() { }
+
+    // Step 1: Initialize blueprint contracts (must be called first)
+    function initializePart1(Blueprints memory _blueprints) public initializer {
+        blueprint = _blueprints;
+    }
```

Comment:
Related to https://github.com/ethereum-optimism/optimism/pull/11875#discussion_r1758969458, what we ultimately want is

```solidity
constructor(SuperchainConfig _superchainConfig, ProtocolVersions _protocolVersions);
function initialize(Blueprints memory _blueprints, string memory _release, bool _isLatest, ImplementationSetter[] memory _setters);
```

as the two signatures. If we get stack too deep, we can try combining the 3 release params into a `SetRelease` struct

Reply:
Latest implementation reflects this end state you mentioned.